src/test/unit/tuvx/tuvx_c_api.cpp

```diff
@@ -385,3 +385,340 @@ TEST_F(TuvxCApiTest, CanCreateProfileMap)
   ASSERT_TRUE(IsSuccess(error));
   DeleteError(&error);
 }
+
+TEST_F(TuvxCApiTest, CannotGetConfiguredRadiator)
+{
+  const char* yaml_config_path = "examples/ts1_tsmlt.yml";
+  SetUp(yaml_config_path);
+  Error error;
+  RadiatorMap* radiator_map = GetRadiatorMap(tuvx, &error);
+  ASSERT_TRUE(IsSuccess(error));
+  ASSERT_NE(radiator_map, nullptr);
+  Radiator* radiator = GetRadiator(radiator_map, "foo", &error);
+  ASSERT_FALSE(IsSuccess(error));  // non-host grid
+  ASSERT_EQ(radiator, nullptr);
+  DeleteRadiatorMap(radiator_map, &error);
+  ASSERT_TRUE(IsSuccess(error));
+  DeleteError(&error);
+}
+
+TEST_F(TuvxCApiTest, CanCreateRadiator)
+{
+  Error error;
+  Grid* height = CreateGrid("height", "km", 3, &error);
+  Grid* wavelength = CreateGrid("wavelength", "nm", 2, &error);
+  Radiator* radiator = CreateRadiator("foo", height, wavelength, &error);
+  ASSERT_TRUE(IsSuccess(error));
+  ASSERT_NE(radiator, nullptr);
+
+  // Test for optical depths
+  std::size_t num_vertical_layers = 3;
+  std::size_t num_wavelength_bins = 2;
+  // Allocate array as 1D
+  double* optical_depths_1D = new double[num_wavelength_bins * num_vertical_layers];
+  // Allocate an array of pointers to each row
+  double** optical_depths = new double*[num_vertical_layers];
+  // Fill in the pointers to the rows
+  for (int row = 0; row < num_vertical_layers; row++)
+  {
+    optical_depths[row] = &optical_depths_1D[row * num_wavelength_bins];
+  }
+  int i = 1;
+  for (int row = 0; row < num_vertical_layers; row++)
+  {
+    for (int col = 0; col < num_wavelength_bins; col++)
+    {
+      optical_depths[row][col] = 10 * i;
+      i++;
+    }
+  }
+  SetRadiatorOpticalDepths(radiator, optical_depths[0], num_vertical_layers, num_wavelength_bins, &error);
+  ASSERT_TRUE(IsSuccess(error));
+  for (int row = 0; row < num_vertical_layers; row++)
+  {
+    for (int col = 0; col < num_wavelength_bins; col++)
+    {
+      optical_depths[row][col] = -999.0;
+    }
+  }
+  GetRadiatorOpticalDepths(radiator, optical_depths[0], num_vertical_layers, num_wavelength_bins, &error);
+  ASSERT_TRUE(IsSuccess(error));
+  ASSERT_EQ(optical_depths[0][0], 10.0);
+  ASSERT_EQ(optical_depths[0][1], 20.0);
+  ASSERT_EQ(optical_depths[1][0], 30.0);
+  ASSERT_EQ(optical_depths[1][1], 40.0);
+  ASSERT_EQ(optical_depths[2][0], 50.0);
+  ASSERT_EQ(optical_depths[2][1], 60.0);
+
```

Comment:
I'm not confident whether this is the correct approach. Since the indices are transposed between Fortran and C, this setup of memory allocation is in favor of fortran, so in fortran side we can do indexing as following. @mattldawson, @K20shores, could you review this?
 
```fortran
    optical_depths(:,1) = (/ 30.0, 20.0, 10.0 /)
    optical_depths(:,2) = (/ 70.0, 80.0, 90.0 /)
    call radiator%get_optical_depths( temp_od, error )
    ASSERT_EQ( temp_od(1,1), 30.0 )
    ASSERT_EQ( temp_od(2,1), 20.0 )
    ASSERT_EQ( temp_od(3,1), 10.0 )
    ASSERT_EQ( temp_od(1,2), 70.0 )
    ASSERT_EQ( temp_od(2,2), 80.0 )
    ASSERT_EQ( temp_od(3,2), 90.0 )
```

Reply:
yes, I think this is the best solution. The indexing is different, but the memory should be laid out in the optimal configuration regardless of how elements are accessed, and looping in C and Fortran should just be set up to access the elements in the optimal order:
```
for(int i=0; i < n_i; ++i)
  for(int j=0; j < n_j; ++j)
    optical_depths[i][j] = ...
```
and:
```
do i = 1, n_i
  do j = 1, n_j
    optical_depths(j,i) = ...
  end do
end do
```

Reply:
That makes sense, thank you